### PR TITLE
🍰 2019/kw22/alpha_data_import

### PR DESCRIPTION
**Pull request by Tirokk**
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-28T16:58:25Z" title="Tuesday, May 28th 2019, 6:58:25 pm +02:00">May 28, 2019</time>_
_Merged <time datetime="2019-05-28T22:43:11Z" title="Wednesday, May 29th 2019, 12:43:11 am +02:00">May 29, 2019</time>_
---

## 🍰 Pullrequest

fixed data import from alpha data:
- include all collections (commented out)
- refactored neo4j import script
- use of .env file for (additional) configurations / configuration overrides
- lots of fiddeling with neo4j cql files and cypher shell

### Issues

- relates https://github.com/Human-Connection/Human-Connection/issues/692

### Todo

- [ ] catalogue all the data to be imported (will not be done in this PR)
- [ ] import the data (will not be done in this PR)

### CODE REVIEW

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/badges.cql [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CALL apoc.load.json('file:/tmp/mongo-export/splits/current-chunk.json') YIELD value as badge
+CALL apoc.load.json("file:${IMPORT_CHUNK_PATH_CQL}") YIELD value as badge
```
Comment (Tirokk) [inline deployment/legacy-migration/maintenance-worker/migration/neo4j/badges.cql:1]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-28T17:33:24Z" title="Tuesday, May 28th 2019, 7:33:24 pm +02:00">May 28, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,4 +1,4 @@
-CALL apoc.load.json('file:/tmp/mongo-export/splits/current-chunk.json') YIELD value as badge
+CALL apoc.load.json("file:${IMPORT_CHUNK_PATH_CQL}") YIELD value as badge
```


I don't think you can expand variables like this. It's a `.cql` file, not a bash script.

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
-do
-  for chunk in /tmp/mongo-export/splits/$collection/*
+# Import collection function defintion
+function import_collection () {
+  for chunk in ${IMPORT_PATH}splits/$1/*
   do
-    mv $chunk /tmp/mongo-export/splits/current-chunk.json
-    echo "Import ${chunk}" && cypher-shell < $SCRIPT_DIRECTORY/$collection.cql
+    mv $chunk ${IMPORT_CHUNK_PATH}
+    NEO4J_COMMAND="$(envsubst '${IMPORT_CHUNK_PATH_CQL}' < $(dirname "$0")/$1.cql)"
```
Comment (Tirokk) [inline deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh:14]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-28T17:34:30Z" title="Tuesday, May 28th 2019, 7:34:30 pm +02:00">May 28, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,17 +1,48 @@
 #!/usr/bin/env bash
 set -e
 
-SECONDS=0
-SCRIPT_DIRECTORY="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-
-echo "MATCH (n) DETACH DELETE n;" | cypher-shell
+# import .env config
+set -o allexport
+source $(dirname "$0")/.env
+set +o allexport
 
-for collection in "badges" "categories" "users" "follows" "contributions" "shouts" "comments"
-do
-  for chunk in /tmp/mongo-export/splits/$collection/*
+# Import collection function defintion
+function import_collection () {
+  for chunk in ${IMPORT_PATH}splits/$1/*
   do
-    mv $chunk /tmp/mongo-export/splits/current-chunk.json
-    echo "Import ${chunk}" && cypher-shell < $SCRIPT_DIRECTORY/$collection.cql
+    mv $chunk ${IMPORT_CHUNK_PATH}
+    NEO4J_COMMAND="$(envsubst '${IMPORT_CHUNK_PATH_CQL}' < $(dirname "$0")/$1.cql)"
```


Or is `envsubst` the command to replace `$VARIABLE` in arbitrary files?

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/badges.cql (hunk 1 above)
Comment (Tirokk) [reply deployment/legacy-migration/maintenance-worker/migration/neo4j/badges.cql:1]: > [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-28T19:05:40Z" title="Tuesday, May 28th 2019, 9:05:40 pm +02:00">May 28, 2019</time>_

---


You can if you use magic like I do here: https://github.com/Human-Connection/Human-Connection/pull/697/files#diff-c6d49070ce711f563a75e723b1e8682cR14

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh (hunk 2 above)
Comment (Tirokk) [reply deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh:14]: > [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-28T19:05:52Z" title="Tuesday, May 28th 2019, 9:05:52 pm +02:00">May 28, 2019</time>_

---


yes it is

**Comment by Tirokk** [comment]
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-28T19:06:53Z" title="Tuesday, May 28th 2019, 9:06:53 pm +02:00">May 28, 2019</time>_

---


I would like to merge this with commented out imports to start the next step to analyse which parts are already correctly imported and which are missing - if you agree on that

File: deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
+#import_collection "invites"
+#import_collection "notifications"
+#import_collection "organizations"
+#import_collection "pages"
+#import_collection "projects"
+#import_collection "settings"
+#import_collection "status"
+#import_collection "systemnotifications"
+#import_collection "userscandos"
+#import_collection "usersettings"
```
Comment (Tirokk) [inline deployment/legacy-migration/maintenance-worker/migration/neo4j/import.sh:46]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-28T22:10:12Z" title="Wednesday, May 29th 2019, 12:10:12 am +02:00">May 29, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,17 +1,48 @@
 #!/usr/bin/env bash
 set -e
 
-SECONDS=0
-SCRIPT_DIRECTORY="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-
-echo "MATCH (n) DETACH DELETE n;" | cypher-shell
+# import .env config
+set -o allexport
+source $(dirname "$0")/.env
+set +o allexport
 
-for collection in "badges" "categories" "users" "follows" "contributions" "shouts" "comments"
-do
-  for chunk in /tmp/mongo-export/splits/$collection/*
+# Import collection function defintion
+function import_collection () {
+  for chunk in ${IMPORT_PATH}splits/$1/*
   do
-    mv $chunk /tmp/mongo-export/splits/current-chunk.json
-    echo "Import ${chunk}" && cypher-shell < $SCRIPT_DIRECTORY/$collection.cql
+    mv $chunk ${IMPORT_CHUNK_PATH}
+    NEO4J_COMMAND="$(envsubst '${IMPORT_CHUNK_PATH_CQL}' < $(dirname "$0")/$1.cql)"
+    echo "Import ${chunk}"
+    echo "${NEO4J_COMMAND}" | "${IMPORT_CYPHERSHELL_BIN}" -u ${NEO4J_USERNAME} -p ${NEO4J_PASSWORD}
   done
-done
+}
+
+# Time variable
+SECONDS=0
+
+# Delete all Neo4J Database content
+echo "Deleting Database Contents"
+echo "MATCH (n) DETACH DELETE n;" | "${IMPORT_CYPHERSHELL_BIN}" -u ${NEO4J_USERNAME} -p ${NEO4J_PASSWORD}
+
+# Import Data
+echo "Start Importing Data"
+import_collection "badges"
+#import_collection "categories"
+#import_collection "comments"
+#import_collection "contributions"
+#import_collection "emotions"
+#import_collection "follows"
+#import_collection "invites"
+#import_collection "notifications"
+#import_collection "organizations"
+#import_collection "pages"
+#import_collection "projects"
+#import_collection "settings"
+#import_collection "shouts"
+#import_collection "status"
+#import_collection "systemnotifications"
+#import_collection "users"
+#import_collection "userscandos"
+#import_collection "usersettings"
```


I cannot approve these changes. I would like to preserve the status quo, which is a lot more than just badges.

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-05-28T22:38:48Z" title="Wednesday, May 29th 2019, 12:38:48 am +02:00">May 29, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/697?src=pr&el=h1) Report
> Merging [#697](https://codecov.io/gh/Human-Connection/Human-Connection/pull/697?src=pr&el=desc) into [master](https://codecov.io/gh/Human-Connection/Human-Connection/commit/3c580c9296ae84423cc3bf8aa35b18cfe8b3952b?src=pr&el=desc) will **increase** coverage by `0.14%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/Human-Connection/Human-Connection/pull/697?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #697      +/-   ##
==========================================
+ Coverage   21.27%   21.41%   +0.14%     
==========================================
  Files         161      161              
  Lines        2257     2284      +27     
  Branches      113      124      +11     
==========================================
+ Hits          480      489       +9     
- Misses       1749     1767      +18     
  Partials       28       28
```


| [Impacted Files](https://codecov.io/gh/Human-Connection/Human-Connection/pull/697?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [webapp/layouts/default.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/697/diff?src=pr&el=tree#diff-d2ViYXBwL2xheW91dHMvZGVmYXVsdC52dWU=) | `0% <0%> (ø)` | :arrow_up: |
| [webapp/components/User/index.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/697/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvVXNlci9pbmRleC52dWU=) | `95.24% <0%> (+2.38%)` | :arrow_up: |
| [webapp/pages/profile/\_id/\_slug.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/697/diff?src=pr&el=tree#diff-d2ViYXBwL3BhZ2VzL3Byb2ZpbGUvX2lkL19zbHVnLnZ1ZQ==) | `3.23% <0%> (+3.23%)` | :arrow_up: |

**Review by Mogge — APPROVED** [review]
OK!

**Pull request closed without merging**